### PR TITLE
feat: Support paying lightning invoices

### PR DESCRIPTION
**Pull request by klochowicz**
In order to test, I have added some basic API to generate an invoice on the maker side.

TODO(follow-up): Improve the invoice generation API and hook it up into 10101.

### CODE REVIEW

**Comment by klochowicz** [comment]
I've tested it by creating an invoice from the maker, and pasted it in into the taker.
result after sending a couple invoices: 
image

note: I have managed to send the same invoice twice, which is probably not ideal.

File: lib/wallet/send.dart [hunk 1]
```diff
@@ -1,17 +1,85 @@
...
+  void initState() {
+    super.initState();
+  }
+
   @override
   Widget build(BuildContext context) {
     return Scaffold(
         appBar: AppBar(
-          title: const Text('Send Payment'),
+          title: const Text('Send Lightning Invoice'),
```
Comment (bonomat) [inline lib/wallet/send.dart:29]: 🙃
That reads weird. Imho you are `paying` an invoice, not `sending` it.

File: lib/wallet/send.dart [hunk 2]
```diff
@@ -1,17 +1,85 @@
...
+                      margin: const EdgeInsets.fromLTRB(0, 0, 20, 30),
+                      child: Column(
+                        mainAxisAlignment: MainAxisAlignment.end,
+                        children: [
+                          Row(
+                            mainAxisAlignment: MainAxisAlignment.end,
+                            children: [
+                              ElevatedButton(
+                                  onPressed: () async {
+                                    FLog.info(text: "Sending invoice" + invoice);
```
Comment (bonomat) [inline lib/wallet/send.dart:61]: Same as above. `Paying` and invoice

File: maker/src/routes.rs [hunk 3]
```diff
@@ -119,3 +121,22 @@ pub async fn post_open_channel(
 
     Ok(Json(OpenChannelResponse { funding_txid }))
 }
+
+#[rocket::post("/invoice/send/<invoice>")]
```
Comment (bonomat) [inline maker/src/routes.rs:125]: I'd also change this `send` to `pay`.
Note, this API is open to the public. I don't mind for the tournament but thought to point it out.

**Review by bonomat — APPROVED** [review]
Looks good. I'd just change the naming a bit.

**Comment by da-kami** [comment]
> I've tested it by creating an invoice from the maker, and pasted it in into the taker.
result after sending a couple invoices:

One thing I noticed is odd in the screenshot is, that it says `Payment Sent` but then it displays `+10` - I think it should be `-10` if we sent it?

**Comment by bonomat** [comment]
> > I've tested it by creating an invoice from the maker, and pasted it in into the taker.
> > result after sending a couple invoices:
> 
> One thing I noticed is odd in the screenshot is, that it says `Payment Sent` but then it displays `+10` - I think it should be `-10` if we sent it?

You might need to adjust it in the frontend. Same for the BTC payments.

File: rust/src/wallet.rs [hunk 4]
```diff
@@ -551,6 +579,7 @@ pub enum Flow {
     Outbound,
 }
 
+// TODO: Remove this? Seems to be exactly the same as HTLCStatus
```
Comment (da-kami) [inline rust/src/wallet.rs:582]: I would not bother :)

**Review by da-kami — APPROVED** [review]

File: rust/src/wallet.rs (hunk 4 above)
Comment (klochowicz) [reply rust/src/wallet.rs:582]: let's keep this until the tournament ends, but then probably needs to go, as this is just adding confusion for no reason :p

**Comment by klochowicz** [comment]
@bonomat I've created a github issue to make sure we can align on the naming after the tournament: https://github.com/itchysats/10101/issues/258

File: maker/src/routes.rs (hunk 3 above)
Comment (klochowicz) [reply maker/src/routes.rs:125]: I've created a new ticket out of this so we won't forget after the tournament.

Comment (klochowicz) [reply maker/src/routes.rs:125]: > I'd also change this `send` to `pay`. Note, this API is open to the public. I don't mind for the tournament but thought to point it out.

that's true. I don't think adding authentication when only using testnet makes much sense, so I didn't add it yet.
However, I added a ticket to ensure we will not forget about it later: 
https://github.com/itchysats/10101/issues/259

**Pull request merged**